engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_week_stream_logs_strategy.go

```diff
@@ -0,0 +1,221 @@
+package stream_logs_strategy
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/enclave"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/logs_clock"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/volume_consts"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/volume_filesystem"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/logline"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	oneWeekDuration = 7 * 24 * time.Hour
+)
+
+// This strategy pulls logs from filesytsem where there is a log file per year, per week, per enclave, per service
+// Weeks are denoted 01-52
+// eg.
+// [.../28/d3e8832d671f/61830789f03a.json] is the file containing logs from service with uuid 61830789f03a, in enclave with uuid d3e8832d671f,
+// in the 28th week of the current year
+type PerWeekStreamLogsStrategy struct {
+	time logs_clock.LogsClock
+}
+
+func NewPerWeekStreamLogsStrategy(time logs_clock.LogsClock) *PerWeekStreamLogsStrategy {
+	return &PerWeekStreamLogsStrategy{
+		time: time,
+	}
+}
+
+func (strategy *PerWeekStreamLogsStrategy) StreamLogs(
+	ctx context.Context,
+	fs volume_filesystem.VolumeFilesystem,
+	logsByKurtosisUserServiceUuidChan chan map[service.ServiceUUID][]logline.LogLine,
+	streamErrChan chan error,
+	enclaveUuid enclave.EnclaveUUID,
+	serviceUuid service.ServiceUUID,
+	conjunctiveLogLinesFiltersWithRegex []logline.LogLineFilterWithRegex,
+	shouldFollowLogs bool,
+) {
+	paths := strategy.getRetainedLogsFilePaths(fs, volume_consts.LogRetentionPeriodInWeeks, string(enclaveUuid), string(serviceUuid))
+	if len(paths) == 0 {
+		streamErrChan <- stacktrace.NewError(
+			`No logs file paths for service '%v' in enclave '%v' were found. 
+					This is a bug, indicating either:
+					1) Logs for the current week are not being stored.
+					2) Logs were manually removed.`,
+			serviceUuid, enclaveUuid)
+		return
+	}
+	if len(paths) < volume_consts.LogRetentionPeriodInWeeks+1 {
+		logrus.Warnf(
+			`We expected to retrieve logs going back '%v' weeks, but instead retrieved logs going back '%v' weeks. 
+					This indicates either:
+					1) The enclave has not been running longer than the log retention period.
```

Comment:
nit: is that really a warn then? I imagine this will be pretty frequent. I'd be more worried though is `len(paths) > volume_consts.LogRetentionPeriodInWeeks+1`